store/store.go

```diff
@@ -181,6 +181,22 @@ func (s *Store) createODSQ4File(
 		)
 	}
 
+	// if file already exists, check if it's corrupted
+	if errors.Is(err, os.ErrExist) {
+		// Validate the size of the file to ensure it's not corrupted
+		err = file.CheckODSQ4Size(pathODS, pathQ4, square)
+		if err != nil {
+			err = s.removeODSQ4(height, roots.Hash())
+			if err != nil {
+				return false, fmt.Errorf("removing corrupted ODSQ4 file: %w", err)
+			}
+			err = file.CreateODSQ4(pathODS, pathQ4, roots, square)
+			if err != nil {
+				return false, fmt.Errorf("recreating ODSQ4 file: %w", err)
+			}
```

Comment:
Readbility nit: extract into a `validateAnRecoverX` method

Reply:
I think it is much easier to understand what is going on here by not hiding details into smaller func. Function calls are self-explanatory here and `validateAnRecoverX` is not going to be reused anywhere

Reply:
The recovery details have layers of nesting involved  with edcase logic unrelated to create func. It would be more readable for a novice to read what simple create does and jump on recovery logic only and only if its needed

Reply:
Essentially recovery logic is a separate operation thus needs a separate method